### DISCUSSION TITLE
Settings for the win

### DISCUSSION THREAD

**Discussion by lewoudar** [pydantic/pydantic#4240] [Pydantic V2]
Hi folks,
First of all I want to say thank you for pydantic, it is now one of the libraries I systematically download in all my new projects 🙂
After reading your blog post for pydantic,I I'm really excited about its release 🤩 but I have two concerns.

## the single decorator validate

Although it looks appealing to have one validator to "rule them all", I'm not sure it is a good idea. I guess, this decorator will contain a lot of arguments to support all the use cases you mentioned in your article. IMHO, it will make it difficult to read/understand and more error prone. I am much in favor to have separate decorators with clear name for each use case like `validate_function`, `validate_dataclass`, `validate_typed_dict`, etc..
Another solution could be to have a single `validate` decorator and create other specialised decorators using the [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial) function.

## Settings management

Please don't remove this! 😭😭
This is one of my two favorite features with the `BaseModel` class. It helps me to respect [one](https://12factor.net/config) of the twelve-factor app principles. 
I don't understand the issue of maintaining it in pydantic, it is not explained in the blog and IMO (yes it is a little biased) it should be kept.

Anyway, I'm looking forward for the V2, thank you again for your work for the python community 😉

**Comment by samuelcolvin** [comment]
Thanks for your feedback.

The validator to rule them all, is just an idea at this stage, we'll see how it turns out.

I agree it can't be too complicated, but I think it's arguments should just be the config settings which will be consistent between use cases.

On settings, they definitely won't be "removed", there's a chance they could be "moved" to a separate project, but since they need to continue to be maintained and compatible, it might actually be more work to move them.

I think I might delay moving settings to V3, it at all.

Whatever happens, installing pydantic-settings will either be a required dependency of pydantic, or an optional dependency installable via `pip install pydantic[settings]`.

**Reply by lewoudar** [reply]
Thank you for the reply,
I will see what the `validate` decorator will look like. Also, I'm glad to read that Settings management will still be there.

**Comment by andrewthetechie** [comment]
I very much like the idea of Settings management being a separate package. I use settings management very frequently in places where I don't need full pydantic and have no other models.

**Reply by samuelcolvin** [reply]
If we make it a separate package, it will depend on pydantic, so full pydantic will have to be installed.

**Reply by andrewthetechie** [reply]
Ahh drat, I guess that makes sense though

**Reply by samuelcolvin** [reply]
Remember also that pydantic V2 will be ~8x smaller to install than pydantic V1 compiled, so it'll already be a lot lighter.

**Reply by andrewthetechie** [reply]
Yeah, I'm looking forward to v2
